wagl/data.py

```diff
@@ -406,6 +406,10 @@ def read_subset(fname, ul_xy, ur_xy, lr_xy, ll_xy, edge_buffer=0, bands=1):
     geobox_subs = GriddedGeoBox(shape=subs.shape, origin=(ul_x, ul_y),
                                 pixelsize=geobox.pixelsize, crs=prj)
 
+    # test for intersection
+    if not geobox_subs.intersects(geobox):
```

Comment:
can you confirm we use read subset for only DEM subset, as this method might be called in other functions.

Reply:
You're right.  It is also used by [brdf](https://github.com/GeoscienceAustralia/wagl/blob/develop/wagl/brdf.py#L311).

Strictly speaking, anything outside of the domain shouldn't attempt to read.  For the cases where there might be a fallback, such as in brdf, then the exception should be caught and handled appropriately.  In the BRDF case, it would still fail for the example of no intersection.
So not really any different, to before, we are no explicit on the return message that the subset doesn't intersect with the geospatial extent we are attempting to read from.

Reply:
With that in mind, I think we're fine.  As the `read_subset` func would've failed for the brdf module if there was no intersection.  This now provides a clearer message, rather than outputting a plain broadcasting index error.